js/ui/appDisplay.js

```diff
@@ -1696,10 +1696,20 @@ const AppFolderPopup = new Lang.Class({
             this.popup();
     },
 
+    _adjustCloseButton: function() {
+        // When the popup has its arrow on top, the close button needs to be
+        // positioned a bit lower so it's placed on the boxPointer corner's edge
+        this._boxPointer.bin.realize();
+        const marginTop = this._boxPointer.bin.get_y();
+        this.closeButton.set_y(this.closeButton.get_y() + marginTop);
```

Comment:
Would it be possible to have this logic moved into a callback for when the 'realize' signal is emitted over `this._boxPointer.bin`? I find it strange that you have to call gtk_widget_realize() explicitly here.